qiskit_experiments/analysis/plotting.py

```diff
@@ -40,7 +40,14 @@ def plot_curve_fit(
     Args:
         func: the fit function for curve_fit.
         result: an AnalysisResult from curve_fit.
-        confidence_interval: if True plot the confidence interval from popt_err.
+        monotonicity: a list or dictionary of Booleans, whose length is the number of parameters.
+           Choice of list or dictionary has to match the format of `result["popt"]`.
+           For every parameter, does the function increase when the parameter increases?
+           If not None, then a confidence interval will be displayed -
+           the plot will be colored in the areas of possible values of the fit function,
+           where possible values are those that can be obtained when every parameter is in the range
+           [fitted_value - error, fitted_value + error]
```

Comment:
It's not really clear here how `error` is computed, and how its computation is relevant to the monotonicity parameter.

Reply:
`error` comes from the fitter. If the function increases with the parameter, then it gets its minimum value when the parameter is equal to `fitted_value - error` and its maximum when the parameter is `fitted_value + error`. If it's decreasing, then the minimum is achieved when the parameter is equal to `fitted_value + error` and the maximum is achieved when the parameter is `fitted_value - error`. If it's not increasing nor decreasing then we don't know when it gets its minimum and maximum.